### ISSUE TITLE
Unable to download maps on Linux

### ISSUE THREAD

**Issue by scooterx3** [ZeroK-RTS/Zero-K#2641]
Chobby's map downloader keeps failing on every attempted map download. I'm not sure what the nitty gritty details are, but I know that recently windows users have been seeing problems with their antivirus killing the downloads of maps or even the client itself when running the installer. I'm on Fedora 25, the only thing I could think of that was related to an antivirus would've been SELinux. I temporarily disabled that, and still can't get any maps to download. I can get them manually though (using wget or just through the browser). 

I asked the Discord room about it, posted my infolog_full.txt and TheMooseIsLoose told me to post my infolog_full.txt here for further investigation. So here it is: [infolog_full.txt](https://github.com/ZeroK-RTS/Zero-K/files/1493635/infolog_full.txt)

**Comment by GoogleFrog** [comment]
This infolog does not look like an infolog of a chobby that managed to start.

**Comment by scooterx3** [comment]
I'll have to try to generate another.

**Comment by Licho1** [comment]
The log looks like firewall block.
Program is unable to connect to server.
Does http://zero-k.info/ work for you?

**Comment by scooterx3** [comment]
Yes, I am able to get to zero-k.info, and I'm not sure why the log looks
like it does, suggesting I can't get in at all. I still have yet to get
another example of what happens when it fails.

Note that sometimes I can't get in because of a firewall restriction,
but in those cases I just set up a couple of SSH tunnels and edit my
/etc/hosts so zero-k.info points to 172.0.0.1 (localhost) and I'm off to
the races. In those cases, previously, the maps downloaded fine as well,
unless I'm not remembering correctly.

The problem I'm having is when I actually am in chobby (normal network,
no tunnels or anything, the network worked fine in the past) and want to
download a new map. It just fails. I'll hopefully be able to upload a
new log file today.

**Comment by scooterx3** [comment]
K, here's the latest one. Seems same as before though. I logged in, joined a game that was using a map that I didn't have, download started, then failed. Then I exited. 
[infolog.txt](https://github.com/ZeroK-RTS/Zero-K/files/1497531/infolog.txt)

**Comment by GoogleFrog** [comment]
Error: [AS::CheckCachedData] found a "/home/riley/zk/maps/ScorpioBattleground.sd7" already in "/home/riley/zk/maps/scorpiobattleground.sd7", ignoring.

What is the file? Maybe it is a failed download.

**Comment by scooterx3** [comment]
Hm, I remember manually downloading that a few days ago, but was 
interested to see that the casing was diferent. I'm fairly sure it's an 
actual map file though.

**Comment by maackey** [comment]
This also happens to me. Its hard to reproduce it reliably -- sometimes the game launches fine, other times it kicks me out of the game and I have to restart the lobby to join the game. You can see by the screenshots that "Download complete" shows, even though I don't have the map.  The past several downloads in the tab are "Connecting..." indefinitely as well. The Downloads tab is completely unresponsive -- clicking cancel does nothing. Only after I restart am I able to download the map and join the game. 


![chobbymapdownloadlinux](https://user-images.githubusercontent.com/3486898/37260380-bd8eee3e-2568-11e8-98bf-5a14522283db.png)
![lastseveraldownloadsconnecting](https://user-images.githubusercontent.com/3486898/37260381-bd9da6d6-2568-11e8-8d7e-e9b77f8ff48d.png)
[infolog.txt](https://github.com/ZeroK-RTS/Zero-K/files/1801025/infolog.txt)

**Comment by RevoluPowered** [comment]
Hello, had the same issue.

The mono run-time must be installed on your platform for this to work properly otherwise this error occurs when trying to download.

```
Zero-K.exe Information: 0 : Chobbyla << DownloadFile {"Name":"Delta Siege Dry Reloaded vBETA4","FileType":"MAP"}
Zero-K.exe Information: 0 : starting download Delta Siege Dry Reloaded vBETA4
Zero-K.exe Error: 0 : Error downloading Delta Siege Dry Reloaded vBETA4: System.DllNotFoundException: libc
  at (wrapper managed-to-native) System.Object:__icall_wrapper_mono_remoting_wrapper (intptr,intptr)
  at (wrapper remoting-invoke) PlasmaShared.IContentService:DownloadFile (string)
  at PlasmaDownloader.Torrents.TorrentDownloader+<>c__DisplayClass5_0.<DownloadTorrent>b__0 () [0x00000] in <41d1d5c12e5644c2993c6d33edc5fe5e>:0 
Zero-K.exe Information: 0 : Chobbyla >> DownloadFileDone {"FileType":"MAP","IsSuccess":false,"IsAborted":false,"Name":"Delta Siege Dry Reloaded vBETA4"}
```

Fixed on arch by running **sudo pacman -S mono** and restarting ZeroK.

**Comment by RevoluPowered** [comment]
Can the dev's please add this as a per-requisite on steam please so that it automatically installs this for normal users.

**Comment by GoogleFrog** [comment]
There is a map pack DLC, but I doubt it is updated. There are too many maps to put them all in the main install. https://github.com/ZeroK-RTS/Zero-K-Infrastructure/issues/2589

**Comment by Licho1** [comment]
You cannot add mono dependency. Just .NET and it has it. In the past steam itself was installing mono on linux for unity games. Is it not so now?

**Comment by RevoluPowered** [comment]
It may be the case that Arch doesn't get that dependency. They officially only support 'ubuntu/debian' based Linux, rather than arch based distributions; even though it works with the actual runtime installed.

At the moment I bet it runs

>  'sudo apt get install mono'

 when you add the .NET requirement. Whereas on arch they use a different package manager command is equivalent

> sudo pacman install -S mono

 the package manager grabs the latest mono rather than messing around with PPA's and other repositories.

Could zerok possibly have a script which warns Linux users if that DLL is not detected for map downloading?

I know it's platform specific fix but Valve aren't planning on fully supporting it across all distros; yet, and users will definitely be using other variants.


**Contacted valve about this too directly with a feature request:**
https://github.com/ValveSoftware/steam-for-linux/issues/6331

**Comment by Anarchid** [comment]
>  In the past steam itself was installing mono on linux for unity games. Is it not so now?

Never was a thing. Unity packs mono using `mkbundle`, same as our Steam version.

**Comment by Licho1** [comment]
wait .. so we mkbundle it.. so that nothing should be needed for linux?

**Comment by Anarchid** [comment]
Steam ZK has a mkbundled chobby that was generated in April 2018 and is probably out of date. 

I have no idea what itch has tbh.

**Comment by RevoluPowered** [comment]
@Licho1 did this ever have a linux release? 

I only ask because on my end, steam is not requiring photon / the compatibility layer for windows games on linux. it just works and updates automatically.

I am thinking because I got a post-release steam key for the game that this may be the source of my problem and it may be looking at the other Linux depot because I'm on that steam key.

does that make any sense? 

I think it is why I had to install mono too... because it's not running on photon its running the Linux build of zerok.

**Comment by Licho1** [comment]
Yeah Zero-K did have linux native release.
It supports linux and windows in it's steam-less mode. 

We also had steam release on linux but it is now diabled and I do not know how to send keys for it..

**Comment by Gyslain** [comment]
Hey guys, today, zk refuses to download the game. Keo has the same issue, we're both on Ubuntu.

![Screenshot from 2022-04-16 17-33-31](https://user-images.githubusercontent.com/11692729/163681173-68bdcea0-8ce5-419a-80a7-4fbb93db90e2.png)

I tried to reboot and to repair local files from Steam, but it didn't work.
Any ideas?

**Comment by Licho1** [comment]
Could you start ZK from steam folder from console and check output?

**Comment by Gyslain** [comment]
I tried ./Zero-K_linux64.sh from steam folder but still doesn't work.
